headers/device.h

```diff
@@ -13,13 +16,16 @@ class Device {
   bool state;
   std::vector<Sensor> sensors;
 
+  const std::string type = "Undefined";
```

Comment:
you're declaring the type as private thus is hidden from the children of the Device class, therefore you need to redefine it in every class, try to define is as protected and assign the value in constructors.
This way you won't need to also redefine getType method in each child